walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,89 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+import java.util.Comparator;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
+  /*
+  * For a particular time slot to work, all attendees must be free to attend the meeting. When a query is made, it will be given a collection of all known events. 
+  * SOLUTION : Check to see if any attendees from the meeting request are already in events for a specfied time, if they are then the timeslot is not avaiable else it is.
+  * Return 
+  */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    /*
+    // sort by start time inplace
+    EventComparator eventComparator = new EventComparator();
+    Collections.sort(events, eventComparator);
+    System.out.print(events);
+    */
+    //System.out.print("EVENTS ARE: " + events);
+
+    //base case : No events, and duration of request is greater than a whole day
+    if( (request.getDuration() > TimeRange.WHOLE_DAY.duration()) && events.isEmpty()) {
+        return Arrays.asList();
+    }
+    // base case: No events
+    if(events.isEmpty()){
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    
+    int start_time = TimeRange.START_OF_DAY;
```

Comment:
It would have been great if you documented what these variables represent. I can clearly see that prevEvent is the event directly before the current event (the one being iterated on), but I don't understand start_time and end_time - in particular, I don't think you need end_time, and it's hard to see how start_time is used because it could be set multiple times before being used.